packages/compass-aggregations/src/components/stage-toolbar/index.tsx

```diff
@@ -104,12 +111,56 @@ export function StageToolbar({
   isCollapsed,
   isDisabled,
   onFocusModeEnableClick,
+  editorRef,
 }: StageToolbarProps) {
   const darkMode = useDarkMode();
   const showFocusMode = usePreference('showFocusMode', React);
 
+  const focusModeButtonRef = useRef<HTMLButtonElement>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [isGuideCueVisible, setIsGuideCueVisible] = useState(false);
+  const [isGuideCueIntersecting, setIsGuideCueIntersecting] = useState(false);
+
+  useEffect(() => {
+    if (!hasSeenFocusModeGuideCue()) {
+      setIsGuideCueVisible(index === 0);
+    }
+  }, [setIsGuideCueVisible, index]);
+
+  useEffect(() => {
+    if (focusModeButtonRef.current) {
+      const observer = new IntersectionObserver(
+        (entries) => {
+          if (entries[0].isIntersecting) {
+            setIsGuideCueIntersecting(true);
+          } else {
+            setIsGuideCueIntersecting(false);
+          }
+        },
+        { threshold: 1 }
```

Comment:
I think we might want this to be a close to zero number instead of 1, otherwise the cue disappears even when the tiniest part of the icon is not on the screen anymore:

![Kapture 2023-02-09 at 19 02 57](https://user-images.githubusercontent.com/5036933/217899891-b93d8fdf-39ae-4ffc-a1a2-3cf60b68810a.gif)